src/jit/codegenarmarch.cpp

```diff
@@ -1539,6 +1543,111 @@ void CodeGen::genCodeForLclFld(GenTreeLclFld* tree)
 }
 
 //------------------------------------------------------------------------
+// genCodeForIndexAddr: Produce code for a GT_INDEX_ADDR node.
+//
+// Arguments:
+//    tree - the GT_INDEX_ADDR node
+//
+void CodeGen::genCodeForIndexAddr(GenTreeIndex* node)
```

Comment:
Given that there are several cases to consider, and that much of this code is comment, do you think it would be useful to put this in codegencommon.cpp, and perhaps even abstract the target-specific bits to avoid `#ifdef`s?